vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt

```diff
@@ -1108,7 +1108,14 @@ class RoomDetailFragment @Inject constructor(
                 showSnackWithMessage(getString(R.string.copied_to_clipboard), Snackbar.LENGTH_SHORT)
             }
             is EventSharedAction.Delete                     -> {
-                roomDetailViewModel.handle(RoomDetailAction.RedactAction(action.eventId, context?.getString(R.string.event_redacted_by_user_reason)))
+                val layout = requireActivity().layoutInflater.inflate(R.layout.dialog_delete_event, null)
+                AlertDialog.Builder(requireActivity())
+                        .setView(layout)
+                        .setPositiveButton(R.string.delete) { _, _ ->
+                            roomDetailViewModel.handle(RoomDetailAction.RedactAction(action.eventId, context?.getString(R.string.event_redacted_by_user_reason)))
```

Comment:
User are complaining that the user language is leaked because of the reason we pass here.
It was not specified in the issue, but we can take the opportunity of this PR to add:
- a checkbox "include a reason", checked by default,
- an TextInputLayout with the reason in the dialog, hint "reason for redacting", pre-filled with R.string.event_redacted_by_user_reason
If the check box is unchecked or the reason is empty, we set null for the reason

See R.layout.dialog_report_content for a similar dialog